app/models/services/tumblr.rb

```diff
@@ -16,12 +16,15 @@ def consumer_secret
   def post(post, url='')
     body = build_tumblr_post(post, url)
     user_info = JSON.parse(client.get("/v2/user/info").body)
-    blogs = user_info["response"]["user"]["blogs"].map { |blog| URI.parse(blog['url']) }
+    blogs = user_info["response"]["user"]["blogs"]
     tumblr_ids = {}
     blogs.each do |blog|
-      resp = client.post("/v2/blog/#{blog.host}/post", body)
-      if resp.code == "201"
-        tumblr_ids[blog.host.to_s] = JSON.parse(resp.body)["response"]["id"]
+      if blog["primary"]
+        blogurl = URI.parse(blog["url"])
+        resp = client.post("/v2/blog/#{blogurl.host}/post", body)
+        if resp.code == "201"
+          tumblr_ids[blog.host.to_s] = JSON.parse(resp.body)["response"]["id"]
+        end
       end
```

Comment:
Point.  The `.each` loop would need to make a return if this were ever to support choosing a subset of blogs to post to, but it can go for now.

Tumblr's API docs imply that fields may be omitted if irrelevant, but don't say outright whether this includes the primary field - and the only account to which I have access for testing already has multiple blogs, so no help there.